nucypher/characters.py

```diff
@@ -219,7 +223,10 @@ def select_teacher_nodes(self):
     def cycle_teacher_node(self):
         if not self.teacher_nodes:
             self.select_teacher_nodes()
-        self._current_teacher_node = self.teacher_nodes.pop()
+        try:
+            self._current_teacher_node = self.teacher_nodes.pop()
+        except IndexError:
+            raise self.NotEnoughUrsulas("Don't have enough nodes to select a good teacher.  This is nearly an impossible situation - do you have seed nodes defined?  Is your network connection OK?")
```

Comment:
What are the conditions of escaping the learning loop process? We need to build in a timeout, I think.

Reply:
:heavy_check_mark: